v2/devdoc/log_context_requirements.md

```diff
@@ -0,0 +1,152 @@
+
+**SRS_LOG_CONTEXT_01_019: [** If 2 properties have the same name for a context a compiler error shall be emitted. **]**
+
+## LOG_CONTEXT_STRING_PROPERTY
+
+```c
+#define LOG_CONTEXT_STRING_PROPERTY(property_name, format, ...) \
+    ...
+```
+
+`LOG_CONTEXT_STRING_PROPERTY` allows defining a property/value pair of type `ascii_char_ptr` where the value stored in the pair is constructed using `printf`-like formatting.
+
+**SRS_LOG_CONTEXT_01_007: [** `LOG_CONTEXT_STRING_PROPERTY` shall expand to code allocating a property/value pair of type `ascii_char_ptr` and the name `property_name`. **]**
+
+**SRS_LOG_CONTEXT_01_008: [** `LOG_CONTEXT_STRING_PROPERTY` shall expand to code that stores as value a string that is constructed using `printf`-like formatting based on `format` and all the arguments in `...`. **]**
+
+## LOG_CONTEXT_NAME
+
+```c
+#define LOG_CONTEXT_NAME(log_context_name) \
+    ...
+```
+
+`LOG_CONTEXT_NAME` allows defining the name for a context.
+
+**SRS_LOG_CONTEXT_01_009: [** `LOG_CONTEXT_NAME` shall be optional. **]**
+
+**SRS_LOG_CONTEXT_01_010: [** If `LOG_CONTEXT_NAME` is not used the name for the context shall be empty string. **]**
+
+**SRS_LOG_CONTEXT_01_011: [** If `LOG_CONTEXT_NAME` is specified multiple times a compiler error shall be emitted. **]**
+
+## LOG_CONTEXT_CREATE
+
+```c
+#define LOG_CONTEXT_LOCAL_DEFINE(dest_log_context, parent_context, ...) \
+    ...
+```
+
+`LOG_CONTEXT_LOCAL_DEFINE` allows defining a stack log context in a similar way to how dynamically allocated contexts are defined, but without having a heap allocation.
+
+Note: The number of properties that can be contained in a stack context and the amount of data is limited in order to not use too much of the stack.
+
+`...` is a list of `LOG_CONTEXT_PROPERTY`, `LOG_CONTEXT_STRING_PROPERTY` or `LOG_CONTEXT_NAME` entries.
+
+**SRS_LOG_CONTEXT_01_015: [** `LOG_CONTEXT_LOCAL_DEFINE` shall store one property/value pair that with a property type of `struct` with as many fields as the total number of properties passed to `LOG_CONTEXT_LOCAL_DEFINE` in the `...` arguments. **]**
+
+**SRS_LOG_CONTEXT_01_017: [** The name of the `struct` property shall be the context name specified by using `LOG_CONTEXT_NAME` (if specified). **]**
+
+**SRS_LOG_CONTEXT_01_016: [** `LOG_CONTEXT_LOCAL_DEFINE` shall store the property types and values specified by using `LOG_CONTEXT_PROPERTY` in the context. **]**
+
+**SRS_LOG_CONTEXT_01_018: [** If `parent_context` is non-`NULL`, the created context shall include all the property/value pairs of `parent_context`. **]**
+
+**SRS_LOG_CONTEXT_01_024: [** If the number of properties to be stored in the log context exceeds `LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT`, an error shall be reported by calling `log_internal_error_report` and no properties shall be stored in the context. **]**
```

Comment:
>an error shall be reported by calling `log_internal_error_report` [](http://example.com/codeflow?start=141&length=65)

likely a compiler error can be emitted straight then and there and the compilation can stop. Maybe. Challenge!

Reply:
I take it back, as it inherits unknown quantities from parent (if it has one) :(

Reply:
Yeah, sorry.